Back-End/VaraticPrim/VaraticPrim.Framework/Managers/LocationManager.cs

```diff
@@ -0,0 +1,123 @@
+﻿using AutoMapper;
+using Microsoft.Extensions.Logging;
+using VaraticPrim.Domain.Entity;
+using VaraticPrim.Framework.Exceptions;
+using VaraticPrim.Framework.Models.LocationModels;
+using VaraticPrim.Framework.Models.UserModels;
+using VaraticPrim.Repository.Repository;
+
+namespace VaraticPrim.Framework.Managers;
+
+public class LocationManager
+{
+    private readonly ILocationRepository _locationRepository;
+    private readonly IUserRepository _userRepository;
+    private readonly IMapper _mapper;
+    private readonly ILogger<LocationManager> _logger;
+
+    public LocationManager(
+        IUserRepository userRepository,
+        ILocationRepository locationRepository,
+        IMapper mapper,
+        ILogger<LocationManager> logger)
+    {
+        _userRepository = userRepository;
+        _logger = logger;
+        _locationRepository = locationRepository;
+        _mapper = mapper;
+    }
+
+    public async Task<LocationModel> Create(LocationCreateModel location)
+    {
+        try
+        {
+            _logger.LogInformation("Creating location...");
+
+            var locationEntity = _mapper.Map<LocationEntity>(location);
+
+            await _locationRepository.Insert(locationEntity);
+
+            var validLocationModel = _mapper.Map<LocationModel>(locationEntity);
+            var userEntity = await _userRepository.GetById(locationEntity.UserId);
```

Comment:
what if user not fount? validate if user exists before inserting in database

Reply:
and you dont need to get user by id, you should use the user from location entity created

Reply:
resolve conversation